### PR TITLE
[coverity] fix coverity issue

### PR DESCRIPTION
**Pull request by EunjuYang**
- This commit fixes some coverity issues of
	- AUTO_CAUSES_COPY
	- MSSING_LOCK

Self-evaluation:

1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/dataset/iteration_queue.cpp [hunk 1]
```diff
@@ -107,6 +107,7 @@ ScopedView<Iteration> IterationQueue::requestFilledSlot() {
     auto stop_request_state = FlowState::FLOW_STATE_STOP_REQUESTED;
     bool exchange_result = flow_state.compare_exchange_strong(
       stop_request_state, FlowState::FLOW_STATE_STOPPED);
+    std::scoped_lock lg(empty_mutex);
```
Comment (djeong20) [inline nntrainer/dataset/iteration_queue.cpp:110]: We should consider whether this lock is necessary for this function.
As far as I know, this data race condition occurs due to variables access in `NNTR_THROW_IF` (e.g., `num_being_filled`). 
While `empty_mutex` is used to request empty slots, I don't think it is preferable for `requestFilledSlot` to hold two mutex locks just to print an error message. (please let me know if I'm wrong)

Comment (djeong20) [reply nntrainer/dataset/iteration_queue.cpp:110]: One simple solution is to remove the critical section by changing the error message.

Comment (EunjuYang) [reply nntrainer/dataset/iteration_queue.cpp]: I updated the code you mentioned ([a0b37ca](https://github.com/nnstreamer/nntrainer/pull/2512/commits/a0b37ca481f18c78e453b6b2c378358967234666)). Thanks for your comment! :)

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by SeoHyungjun — APPROVED** [review]
LGTM

**Pull request merged**